common/src/main/java/com/viaversion/viaversion/protocols/base/InitialBaseProtocol.java

```diff
@@ -123,6 +123,16 @@ protected void registerPackets() {
                 wrapper.set(Types.VAR_INT, 0, serverProtocol.getOriginalVersion());
             }
 
+            // Send client intention into the pipeline in case protocols down the line need to transform it
+            try {
+                final List<Protocol> protocols = new ArrayList<>(pipeline.pipes());
+                protocols.remove(this);
+                wrapper.apply(Direction.SERVERBOUND, State.HANDSHAKE, protocols);
+            } catch (CancelException e) {
+                wrapper.user().setActive(false);
```

Comment:
Should maybe just ignore cancellation?

Reply:
I looked at the if (versionProvider == null) code